packages/core-data/src/footnotes/index.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Internal dependencies
+ */
+import getFootnotesOrder from './get-footnotes-order';
+
+let oldFootnotes = {};
+
+export function updateFootnotesFromMeta( blocks, meta ) {
+	const output = { blocks };
+	if ( ! meta ) return output;
```

Comment:
this syntax makes me so nervous. it's been the vector of some of the biggest attacks and looks otherwise so innocent.

Reply:
Can you clarify? Is it the lack of braces around the THEN block?

Reply:
yeah that's right. it's essentially the same discussion as with automatic semicolon insertion, which the industry has overwhelmingly rejected. I thought we overwhelmingly rejected code blocks without braces after major exploits (notably the OpenSSL-bypass).

even `if ( ! meta ) { return output; }` retains the safety that `if ( ! meta ) return output;` eschews, so giving up the safety doesn't even avoid any lines.

there are some trivial cases, particularly with comments, that can fool auto-formatters into creating the bugs. formatters move comments around, and the newlines and comments themselves can rip apart these coupled blocks. the braces tie them together.

curious what the advantages are of doing it this way